src/convertToObject.js

```diff
@@ -7,6 +7,47 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+
+  let cleanedString = '';
+  let insideValue = false;
+
+  for (let i = 0; i < sourceString.length; i++) {
+    const char = sourceString[i];
+
+    if (insideValue) {
+      cleanedString += char;
+
+      if (char === ';') {
+        insideValue = false;
+      }
+    } else {
+      if (char === '\n' || char === '\t') {
+        continue;
+      } else if (char === ':') {
+        insideValue = true;
+      }
+
+      cleanedString += char;
+    }
+  }
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. Consider using `split`, `map`, and `filter` to process the string and build the object.